addons/mail/models/res_users.py

```diff
@@ -31,7 +31,30 @@ class Users(models.Model):
         'Notification Management', required=True, default='email',
         help="Policy on how to handle Chatter notifications:\n"
              "- Handle by Emails: notifications are sent to your email address\n"
-             "- Handle in Odoo: notifications appear in your Odoo Inbox")
+             "- Handle in Odoo: notifications appear in your Odoo Inbox"
+             "- Emails: notifications are sent to your email\n"
+             "- Odoo: notifications appear in your Odoo Inbox")
+    is_moderator = fields.Boolean(string="Is moderator", compute="_compute_is_moderator")
+    messages_to_moderate_count = fields.Integer(string="Count of messages to moderate", compute="_compute_messages_to_moderate_count")
+    moderated_channel_ids = fields.Many2many('mail.channel', 'mail_channel_moderator_rel', string="User moderated channels")
+
+    @api.multi
+    def _compute_is_moderator(self):
+        for user in self:
+            user.is_moderator = bool(self.env['mail.channel'].search_count([
+                    ('moderation', '=', True),
+                    ('moderator_ids', 'in', user.id)
+                ]))
+
+    @api.multi
+    def _compute_messages_to_moderate_count(self):
+        for user in self:
+            user.messages_to_moderate_count = self.env['mail.message'].search_count([
```

Comment:
I fear this one will be a performance killer. I think an hand-made SQL query will be necessary. Indeed mail_message is a model with complex access rights done by hand. In this case I think directly writing an SQL query will be necessary to optimize a bit performances.